cdap-common/src/main/java/io/cdap/cdap/common/service/AbstractRetryableScheduledService.java

```diff
@@ -178,9 +182,9 @@ protected final void runOneIteration() throws Exception {
         }
         this.delayMillis = delayMillis;
       }
-    } catch (Exception e) {
-      LOG.error("Aborting service {} due to non-retryable error", getServiceName(), e);
-      throw e;
+    } catch (Throwable t) {
+      LOG.error("Aborting service {} due to non-retryable error", getServiceName(), t);
```

Comment:
usually we do not log and throw - https://github.com/cdapio/hydrator-plugins/pull/1072#discussion_r406868886

Reply:
This is what this PR is about. When an `Error` is thrown up in this case, the exception is not getting log at all and the service just terminate. The guava service only log `Exception`.